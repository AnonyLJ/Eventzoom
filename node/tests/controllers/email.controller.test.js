import request from 'supertest';
import index from '../../src/root';
import userService from '../../src/services/user.service';
import seriesService from '../../src/services/series.service';
import emailService from '../../src/services/email.service';

describe('testing emails', () => {
	it('should send successfully', async () => {
		userService.getAllUsers = jest.fn().mockImplementation(() => [
			[{ events: ['qwerty'] }],
			[{ events: [] }],
		]);
		seriesService.getUserSubscriptions = jest.fn().mockImplementation((stuff) => stuff);
		emailService.sendEmail = jest.fn();
		const res = await request(index.app)
			.get('/emails')
			.set('Secret', 'cats')
			.send();
		await expect(res.body).toEqual({ success: true });
		await expect(userService.getAllUsers.mock.calls).toMatchSnapshot();
		await expect(seriesService.getUserSubscriptions.mock.calls).toMatchSnapshot();
		await expect(emailService.sendEmail.mock.calls).toMatchSnapshot();
	});
});
