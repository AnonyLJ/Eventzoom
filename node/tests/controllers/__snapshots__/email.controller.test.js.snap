// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing emails should send successfully 1`] = `
Array [
  Array [],
]
`;

exports[`testing emails should send successfully 2`] = `
Array [
  Array [
    Array [
      Object {
        "events": Array [
          "qwerty",
        ],
      },
    ],
  ],
  Array [
    Array [
      Object {
        "events": Array [],
      },
    ],
  ],
]
`;

exports[`testing emails should send successfully 3`] = `
Array [
  Array [
    undefined,
    "subscriptions",
    Object {
      "series": Array [
        Object {
          "events": Array [
            "qwerty",
          ],
        },
      ],
    },
  ],
  Array [
    undefined,
    "subscriptions",
    Object {
      "series": Array [
        Object {
          "events": Array [],
        },
      ],
    },
  ],
]
`;
